### ISSUE TITLE
MethodWriter generated code fails to compile when method has return type

### ISSUE THREAD

**Issue by JerryShea** [OpenHFT/Chronicle-Wire#201]
```
.../PricerOutInterceptingTextMethodWriter.java:89: error: incompatible types: missing return value
        if (! this.updateInterceptor.update("ignoreMethodBasedOnFirstArg", arg1)) return;
 ```

**Comment by RobAustin** [comment]
@JerryShea do you have a failing test you can send me ?

**Comment by RobAustin** [comment]
I've created the following test cases

```
package net.openhft.chronicle.wire;

import junit.framework.TestCase;
import net.openhft.chronicle.bytes.Bytes;

import static net.openhft.chronicle.wire.WireType.BINARY;

public class GenerateMethodWriterTest extends TestCase {

    interface NoReturnType {
        void x(String x);
    }

    interface WithNoReturnType {
        int x(String x);
    }

    public void testUpdateInterceptorNoReturnType() {

        BINARY.apply(Bytes.elasticByteBuffer())
                .methodWriterBuilder(NoReturnType.class)
                .updateInterceptor((methodName, t) -> true)
                .build()
                .x("hello world");
    }

    public void testUpdateInterceptorWithReturnType() {
        BINARY.apply(Bytes.elasticByteBuffer())
                .methodWriterBuilder(WithNoReturnType.class)
                .updateInterceptor((methodName, t) -> true)
                .build()
                .x("hello world");
    }
}
```

**Comment by JerryShea** [comment]
hey Rob, we need to be able to support an interface like https://github.com/ChronicleEnterprise/Chronicle-EFX/blob/1b956081659794b9b5ff2ebf124ab17917afe0bb/common/src/main/java/software/chronicle/efx/common/listener/LadderByQtyListener.java which has a `boolean ignoreMethodBasedOnFirstArg` method.
We can exclude this method from the method writer - Wire only calls it as part of method reader, but it does need code-generated and not fall back to the Proxy mechanism.

**Comment by JerryShea** [comment]
See net.openhft.chronicle.wire.MethodFilterOnFirstArg

**Comment by RobAustin** [comment]
yes, we can change the code to ignore all `defaults` methods that don't have a return type as a `void` or `interface` ( this will allow the code you provided to compile ). However strictly speaking, this method you provided, should not really be on the method writer interface, as "ignoreMethodBasedOnFirstArg" is only required when reading.

**Comment by JerryShea** [comment]
Yes I guess the xxxListener strictly speaking does not have to have the
ignoreMethodBasedOnFirstArg on it.
Could we/should we make a special case for ignoreMethodBasedOnFirstArg like
method readers do?

**Comment by RobAustin** [comment]
I have added the following test

```
    @Test
    public void testUpdateInterceptorWithLadderByQtyListener() {
        BINARY.apply(elasticByteBuffer())
                .methodWriterBuilder(LadderByQtyListener.class)
                .updateInterceptor((methodName, t) -> true)
                .build()
                .ladderByQty("a ladder");
    }


    public interface LadderByQtyListener {
        void ladderByQty(String ladder);

        default void lbq(String name, String ladder) {
            ladderByQty(ladder);
        }

        default boolean ignoreMethodBasedOnFirstArg(String methodName, String ladderDefinitionName) {
            return false;
        }
    }
```
and fixed this issue

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.29](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.29), [BOM-2.20.56](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.56)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
